bundle/edu.gemini.lchquery.servlet/src/main/scala/edu/gemini/lchquery/servlet/LchQueryFunctor.scala

```diff
@@ -195,14 +195,15 @@ class LchQueryFunctor(queryType: LchQueryFunctor.QueryType,
                       case lst if lst.nonEmpty => lst.asScalaList
                     } foreach { tgts =>
                       setTargetsNode(new TargetsNode() {
-                        tgts.map(tgt => makeTargetNode(tgt, env)).foreach(getTargets.add)
+                        tgts.map(tgt => makeTargetNode(tgt, env)).collect {
```

Comment:
If you have a `List[Option[A]]` you can call `.flatten` with the stdlib collections because there is an implicit conversion from `Option` to `Iterable` and `CanBuildFrom` can gloss over the difference between `List` and `Iterable`. But it's a very specific case.

In the general case it's bogus, which is why I prefer `collect`.